presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPushdownUtils.java

```diff
@@ -163,6 +185,56 @@ public ExpressionType getExpressionType()
         return nodeBuilder.build();
     }
 
+    /**
+     * Try to push down query like: `SELECT count(distinct $COLUMN) FROM myTable` to Pinot as `SELECT distinctCount($COLUMN) FROM myTable`.
+     * This function only handles the case of an AggregationNode (COUNT on $COLUMN) on top of an AggregationNode(of non-aggregate on $COLUMN).
+     *
+     * @param nodeBuilder
+     * @param aggregationNode
+     * @param outputColumn
+     * @param aggregation
+     * @return true if push down successfully otherwise false.
+     */
+    private static boolean handlePushDownSingleDistinctCount(ImmutableList.Builder<AggregationColumnNode> nodeBuilder, AggregationNode aggregationNode, VariableReferenceExpression outputColumn, AggregationNode.Aggregation aggregation)
+    {
+        if (aggregation.getCall().getDisplayName().equalsIgnoreCase(COUNT_FUNCTION_NAME)) {
+            List<RowExpression> arguments = aggregation.getCall().getArguments();
+            if (arguments.size() == 1) {
+                RowExpression aggregationArgument = arguments.get(0);
+                // Handle the case of Count Aggregation on top of a Non-Agg GroupBy Aggregation.
+                if (aggregationNode.getSource() instanceof AggregationNode) {
+                    AggregationNode sourceAggregationNode = (AggregationNode) aggregationNode.getSource();
+                    Set<String> sourceAggregationGroupSet = getGroupKeys(sourceAggregationNode.getGroupingKeys());
+                    Set<String> aggregationGroupSet = getGroupKeys(aggregationNode.getGroupingKeys());
+                    aggregationGroupSet.add(aggregationArgument.toString());
+                    if (sourceAggregationGroupSet.containsAll(aggregationGroupSet) && aggregationGroupSet.containsAll(sourceAggregationGroupSet)) {
+                        //(sourceAggregationNode.getGroupingKeys().containsAll(aggregationNode.getGroupingKeys()))
+                        //aggregationArgument.equalsIgnoreCase(sourceAggregationNode.getGroupingKeys().get(0).getName()))
+                        nodeBuilder.add(
+                                new AggregationFunctionColumnNode(
+                                        outputColumn,
+                                        new CallExpression(
+                                                DISTINCT_COUNT_FUNCTION_NAME,
+                                                aggregation.getFunctionHandle(),
+                                                BigintType.BIGINT,
+                                                ImmutableList.of(aggregationArgument))));
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private static Set<String> getGroupKeys(List<VariableReferenceExpression> groupingKeys)
```

Comment:
Seems like a helpful helper function. Can it be used elsewhere too ?

Reply:
I will make it public then, will see if other place could use it also.